src/ucs/sys/event_set.c

```diff
@@ -69,32 +73,55 @@ static inline int ucs_event_set_map_to_events(int raw_events)
     return events;
 }
 
-ucs_status_t ucs_event_set_create(ucs_sys_event_set_t **event_set_p)
+static ucs_sys_event_set_t *ucs_event_set_alloc(int event_fd, unsigned flags)
 {
     ucs_sys_event_set_t *event_set;
-    ucs_status_t status;
 
     event_set = ucs_malloc(sizeof(ucs_sys_event_set_t), "ucs_sys_event_set");
     if (event_set == NULL) {
         ucs_error("unable to allocate memory ucs_sys_event_set_t object");
-        status = UCS_ERR_NO_MEMORY;
-        goto out_create;
+        return NULL;
+    }
+
+    event_set->flags = flags;
+    event_set->epfd  = event_fd;
+    return event_set;
+}
+
+ucs_status_t ucs_event_set_create_from_fd(ucs_sys_event_set_t **event_set_p,
+                                          int event_fd)
+{
+    *event_set_p = ucs_event_set_alloc(event_fd,
+                                       UCS_SYS_EVENT_SET_EXTERNAL_EVENT_FD);
+    if (*event_set_p == NULL) {
+        return UCS_ERR_NO_MEMORY;
     }
 
+    return UCS_OK;
+}
+
+ucs_status_t ucs_event_set_create(ucs_sys_event_set_t **event_set_p)
+{
+    ucs_status_t status;
+    int event_fd;
+
     /* Create epoll set the thread will wait on */
-    event_set->epfd = epoll_create(1);
-    if (event_set->epfd < 0) {
+    event_fd = epoll_create(1);
+    if (event_fd < 0) {
         ucs_error("epoll_create() failed: %m");
-        status = UCS_ERR_IO_ERROR;
-        goto err_free;
+        return UCS_ERR_IO_ERROR;
+    }
+
+    *event_set_p = ucs_event_set_alloc(event_fd, 0);
```

Comment:
Do we need an alias? ex) `UCS_SYS_EVENT_SET_NONE` or something?

Reply:
I don't think so, it's ok to have `0` as we use `unsigned` type

Reply:
My point was it is easier than 0 when someone reading code.
But It's not important.  LGTM